File: homework01/caesar.go
package caesar

func EncryptCaesar(plaintext string) string {
	ciphertext := ""
	for i := 0; i < len(plaintext); i++ {
		symbol := int(plaintext[i])
		if (int('A') <= symbol) && (symbol <= int('Z')) || (int('a') <= symbol && symbol <= int('z')) {
			symbol += 3
			if (symbol > int('Z') && symbol < int('a')) || symbol > int('z') {
				symbol -= 26
			}
			ciphertext += string(symbol)
		} else {
			ciphertext += string(plaintext[i])
		}
	}
	return ciphertext
}

func DecryptCaesar(ciphertext string) string {
	plaintext := ""
	for i := 0; i < len(ciphertext); i++ {
		symbol := int(ciphertext[i])
		if (int('A') <= symbol) && (symbol <= int('Z')) || (int('a') <= symbol && symbol <= int('z')) {
			symbol -= 3
			if (symbol > int('Z') && symbol < int('a')) || symbol < int('A') {
				symbol += 26
			}
			plaintext += string(symbol)
		} else {
			plaintext += string(ciphertext[i])
		}
	}
	return plaintext
}

File: homework01/caesar.py
def encrypt_caesar(plaintext):
    """
    >>> encrypt_caesar("PYTHON")
    'SBWKRQ'
    >>> encrypt_caesar("python")
    'sbwkrq'
    >>> encrypt_caesar("Python3.6")
    'Sbwkrq3.6'
    >>> encrypt_caesar("")
    ''
    """
    ciphertext = ""
    for symbol in plaintext:
        if 'A' <= symbol <= 'Z' or 'a' <= symbol <= 'z':
            symbol_code = ord(symbol) + 3
            if symbol_code > ord('Z') and symbol_code < ord('a') or symbol_code > ord('z'):
                symbol_code -= 26
            ciphertext += chr(symbol_code)
        else:
            ciphertext += symbol
    return ciphertext


def decrypt_caesar(ciphertext):
    """
    >>> decrypt_caesar("SBWKRQ")
    'PYTHON'
    >>> decrypt_caesar("sbwkrq")
    'python'
    >>> decrypt_caesar("Sbwkrq3.6")
    'Python3.6'
    >>> decrypt_caesar("")
    ''
    """
    plaintext = ""
    for symbol in ciphertext:
        if 'A' <= symbol <= 'Z' or 'a' <= symbol <= 'z':
            symbol_code = ord(symbol) - 3
            if symbol_code > ord('Z') and symbol_code < ord('a') or symbol_code < ord('A'):
                symbol_code += 26
            plaintext += chr(symbol_code)
        else:
            plaintext += symbol

    return plaintext
